demo/src/mpsc_registry.rs

```diff
@@ -0,0 +1,80 @@
+use std::{collections::HashMap, fmt::Debug, sync::Arc};
+
+use tokio::sync::{mpsc, RwLock};
+
+#[derive(Clone)]
+pub struct MpscRegistry<T: Debug> {
+    sending: Arc<RwLock<HashMap<String, mpsc::Sender<T>>>>,
+    receiving: Arc<RwLock<HashMap<String, mpsc::Receiver<T>>>>,
+}
```

Comment:
Wasn't the previous implementation similar to this in terms of communication? Maybe I'm missing something but I don't get why the shared ownership and interior mutability is needed. 

Each agent should be in charge of its own connections (senders to other agents and receivers from other agents) and practically own them exclusively. Or is there some reason why this can't be achieved?